### PR TITLE
[SofaGeneralAnimationLoop] Accumulate matrix entries in a triplet list

### PR DESCRIPTION
**Pull request by alxbilger**
Based on https://github.com/sofa-framework/sofa/pull/2394

Previously, a `CompressedRowSparseMatrix` accumulated matrix entries and was then converted to Eigen. The conversion gets triplets and detects duplicates and does the compression. Therefore, all the work to build a matrix is done in the conversion. This work is not required during the accumulation as it is done later in any case.

I propose to accumulate all the matrix entries in a simple list. This accumulation does nothing else than inserting a new element in the vector. This should be faster than using a `CompressedRowSparseMatrix`.

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

File: SofaKernel/modules/Sofa.LinearAlgebra/src/sofa/linearalgebra/TripletMatrix.h [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+/**
+ * Store matrix entries as a list of triplets containing:
+ * - the index of the row
+ * - the index of the column
+ * - the value at the location pointed by both indices
+ *
+ * This class is not meant to be used as a matrix, but rather as a proxy of a BaseMatrix to build a triplet list and
+ * provide it later to some processing (in Eigen SparseMatrix::setFromTriplets for example).
+ *
+ * Duplicates are not removed.
```
Comment (jnbrunet) [inline SofaKernel/modules/Sofa.LinearAlgebra/src/sofa/linearalgebra/TripletMatrix.h:43]: Maybe a small note here that duplicates in Eigen are usually summed up (if I remember correctly)

**Comment by jnbrunet** [comment]
Hey @alxbilger , just a small question, did you try it with large K and J? Does it makes you RAM sweat with all those triplets stored ?

Alt Text

File: SofaKernel/modules/Sofa.LinearAlgebra/src/sofa/linearalgebra/TripletMatrix.h (hunk 1 above)
Comment (alxbilger) [reply SofaKernel/modules/Sofa.LinearAlgebra/src/sofa/linearalgebra/TripletMatrix.h:43]: You are right

**Comment by alxbilger** [comment]
> Hey @alxbilger , just a small question, did you try it with large K and J? Does it makes you RAM sweat with all those triplets stored ?
> 

Actually, I am not even sure that the performances are better in any case, specially for large K or J. I admit I did not benchmark for multiple mesh size. I'll put this PR back to WIP and consider your remark on the RAM (but I am confident it's OK). I'll need to revert this PR from #2411.

**Pull request closed without merging**